x-pack/plugins/alerting/server/saved_objects/index.ts

```diff
@@ -85,6 +88,24 @@ export type RuleAttributesNotPartiallyUpdatable =
   | 'meta'
   | 'alertDelay';
 
+export const AdHocRunAttributesToEncrypt = ['apiKeyToUse'];
+export const AdHocRunAttributesIncludedInAAD = [
+  'enabled',
+  'start',
+  'duration',
+  'createdAt',
+  'rule',
+  'spaceId',
+];
```

Comment:
Just FYI - once these attributes are included in AAD, they cannot be removed due to the limitations of zero-downtime upgrades in serverless (at least for the foreseeable future). I just wanted to double check that each one was considered preferred/necessary to include in AAD before approving the PR.

Reply:
Thanks for flagging @jeramysoucy! I discussed it with the team and we're going to start with the minimum fields that make sense at this time. Updated in https://github.com/elastic/kibana/pull/177622/commits/23cda6db82d06dfe7004b20f08a085e4159d7646

If we want to include additional fields in the future, what is the process for that?

Reply:
@ymao1 Unfortunately, adding or removing an **existing** attribute to/from AAD is not supported at this time. I wish I had a better answer than that. New attributes can be added to AAD, but only before they are utilized/populated:

> The addition will require 2 serverless release stages.
> Release 1: Add the field to attributesToIncludeInAAD in the ESO type registration. Do not yet use/populate the new field.
> Release 2: Begin using the new field. Implement model version change to backfill data as needed.

Adding new nested fields of attributes that are already included in AAD is handled automatically.

We have a backlog task to document developer guidelines for AAD, as it is not straight-forward. Additionally, our team has a longterm goal of supporting any AAD change in serverless, but this work is not in our roadmap at this time.

For now, the best approach is to be conservative with which attributes are included in AAD. You can always tag the Kibana Security team for advice or input when making ESO changes. It is our intention that down the road we can make ESO management much easier for developers.